### ISSUE TITLE
Incomplete migration info for custom ajv $ref shared schema handling

### ISSUE THREAD

**Issue by lmeyerov** [fastify/fastify#2429]
## 🐛 Bug Report

The migration docs for supplying a custom ajv instance are pretty opaque around supporting $ref for shared schema. It seems, overall, the idea is fastify can no longer resolve through a shared ajv, so you need to skip fastify more than the docs make it seem. In particular, in addition to

```
fastify.setValidatorCompiler(({schema}) => { return ajv.compile(schema);  })
```

We ended up also adding:

```
fastify.decorate('_customAddSchema', function (s) { ajv.addSchema(s); return this });
fastify.decorate('_customGetSchema', (ref) => ajv.getSchema(ref));
``` 

And then rewrote:

```
{'x': 'x#'} => {'x': fastify._customGetSchema('http://site/schema/common.json#x')}
{'x': {$ref: 'http://site/schema/common.json#x'}} => {'x': fastify._customGetSchema('http://site/schema/common.json#x')}
```
and
```
fastify.addSchema(x) => fastify._customSetSchema(x)
```

Improving the migration guidance would help -- it isn't obvious whether this is safe & recommended for solving the issue. (I'd prefer ajv integration so this could be skipped, but #2023 sounds like that was problematic?)

## Your Environment

- *node version*: 14
- *fastify version*: 3.X

**Comment by Eomm** [comment]
When you customize the validator compiler, all is under the control of the dev and fastify doesn't know anything about what validator the dev has chosen.

> I'd prefer ajv integration so this could be skipped

Customizing `ajv` as a validator, right now, it is not necessary thanks to the [ajv config](https://github.com/fastify/fastify/blob/master/docs/Server.md#ajv).

> fastify.decorate('_customAddSchema', function (s) { ajv.addSchema(s); return this });

This solution works for your use case since it doesn't cover:

- encapsulation
- run `_customAddSchema` after the definition of a route that needs that schema (fastify support this idempotent kind of writing code)

This happens because fastify create the `ajv` instance after the execution of all the `.register` contexts.

I don't think we should improve the migration guideline since this behaviour is the same as fastify v2.
I would clarify that in the [`Validation-and-Serialization.md#using-other-validation-libraries`](https://github.com/fastify/fastify/blob/master/docs/Validation-and-Serialization.md#using-other-validation-libraries) doc for sure

**Comment by jsumners** [comment]
I think this will be solved by the resolution of #2446
